.github/workflows/ci.yaml

```diff
@@ -86,60 +51,156 @@ jobs:
+    runs-on: ubuntu-20.04
+    if: ${{ always() }}
+    needs: [build_job, test_job]
     steps:
-      - name:  Extract job starting & ending time
+      - name:  Extract TEST and BUILD job starting & ending time
         id: execution_timestamps
         run: |
-          echo "::set-output name=starttime::$(curl -H "Accept: application/vnd.github+json" https://api.github.com/repos/${{ github.repository }}/actions/runs/${{ github.run_id }}/jobs | jq '.jobs[0].started_at')" && echo "::set-output name=endtime::$(curl -H "Accept: application/vnd.github+json" https://api.github.com/repos/${{ github.repository }}/actions/runs/${{ github.run_id }}/jobs | jq '.jobs[0].completed_at')"
-      
+          echo "::set-output name=starttime_test::$(curl -H "Accept: application/vnd.github+json" https://api.github.com/repos/${{ github.repository }}/actions/runs/${{ github.run_id }}/jobs | jq '.jobs[0].started_at')" && echo "::set-output name=endtime_test::$(curl -H "Accept: application/vnd.github+json" https://api.github.com/repos/${{ github.repository }}/actions/runs/${{ github.run_id }}/jobs | jq '.jobs[0].completed_at')"
+          echo "::set-output name=starttime_build::$(curl -H "Accept: application/vnd.github+json" https://api.github.com/repos/${{ github.repository }}/actions/runs/${{ github.run_id }}/jobs | jq '.jobs[1].started_at')" && echo "::set-output name=endtime_build::$(curl -H "Accept: application/vnd.github+json" https://api.github.com/repos/${{ github.repository }}/actions/runs/${{ github.run_id }}/jobs | jq '.jobs[1].completed_at')"
+
       - name: calculate time
         id: calculate_time
         run: |
-          started_at=${{steps.execution_timestamps.outputs.starttime}}
-          completed_at=${{steps.execution_timestamps.outputs.endtime}}
+          started_at_test=${{steps.execution_timestamps.outputs.starttime_test}}
+          completed_at_test=${{steps.execution_timestamps.outputs.endtime_test}}
+          started_at_build=${{steps.execution_timestamps.outputs.starttime_build}}
+          completed_at_build=${{steps.execution_timestamps.outputs.endtime_build}}
 
-          START_TIME=$(date -d "$started_at" +%s)
-          END_TIME=$(date -d "$completed_at" +%s)
+          START_TIME_TEST=$(date -d "$started_at_test" +%s)
+          END_TIME_TEST=$(date -d "$completed_at_test" +%s)
+          START_TIME_BUILD=$(date -d "$started_at_build" +%s)
+          END_TIME_BUILD=$(date -d "$completed_at_build" +%s)
 
-          ELAPSE=$(( $END_TIME - $START_TIME ))
+          ELAPSE_TEST=$(( $END_TIME_TEST - $START_TIME_TEST ))
+          ELAPSE_BUILD=$(( $END_TIME_BUILD - $START_TIME_BUILD ))
+          ELAPSE=$(( $ELAPSE_TEST + $ELAPSE_BUILD ))
           echo $ELAPSE
           echo "$(($ELAPSE/60/60))h $(($ELAPSE/60%60))m $(($ELAPSE%60))s"
           echo "::set-output name=elpsd::$ELAPSE"
           echo "::set-output name=el_hours::$(($ELAPSE/60/60))"
           echo "::set-output name=el_minutes::$(($ELAPSE/60%60))"
```

Comment:
that is too much implementation for the duration calculation. Like you went to your local grocery store by flying to Africa and back first.

let's find a better way.

Reply:
found a better way: dedicated steps at the beginning and the end of the workflow to get timestamps.